app/src/main/java/com/kms/cura/view/CheckBoxAdapter.java

```diff
@@ -0,0 +1,98 @@
+package com.kms.cura.view;
+
+import android.content.Context;
+import android.database.DataSetObserver;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.BaseAdapter;
+import android.widget.CheckBox;
+import android.widget.LinearLayout;
+import android.widget.SpinnerAdapter;
+import android.widget.TextView;
+
+import com.kms.cura.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by toanbnguyen on 5/30/2016.
+ */
+public class CheckBoxAdapter extends BaseAdapter {
+    Context context;
+    int resource;
+    ArrayList<String> checkList;
+    boolean[] checked;
+    public CheckBoxAdapter(Context context, int resource, ArrayList objects, boolean[] checked) {
+        this.context = context;
+        this.resource = resource;
+        this.checkList = objects;
+        if (checked != null)
+            this.checked = checked;
+        else {
+            this.checked = new boolean[checkList.size() - 1];
+            for (int i = 0; i < checkList.size() - 1; ++i)
+                this.checked[i] = false;
+        }
+    }
+
+    @Override
+    public int getCount() {
+        return checkList.size() - 1; // not to include the hint in the list
+    }
+
+    @Override
+    public Object getItem(int position) {
+        return checkList.get(position);
+    }
+
+    @Override
+    public long getItemId(int position) {
+        return position;
+    }
+
+    @Override
+    public View getView(final int position, View convertView, ViewGroup parent) {
```

Comment:
Here is my suggestion:
`@Override
    public View getView(final int position, View convertView, ViewGroup parent) {
        convertView = initConvertView(convertView, parent);
        initCheckBoxAndTextViewForPosition(position, convertView);
        return convertView;
    }

    private void initCheckBoxAndTextViewForPosition(final int position, View convertView) {
        String name = checkList.get(position);
        CheckBox cb = (CheckBox) convertView.findViewById(R.id.checkBox);
        final TextView tv = (TextView) convertView.findViewById(R.id.tvHint);
        if (position < checkList.size() - 1) {
            cb.setText(name);
            tv.setVisibility(View.INVISIBLE);
            cb.setChecked(checked[position] || (!checked[position] && cb.isChecked()));
            initListenersForCheckBox(position, cb);
            return;
        }
        if (position == checkList.size() - 1) {
            tv.setText(name);
            cb.setVisibility(View.INVISIBLE);
        }
    }

    private void initListenersForCheckBox(final int position, CheckBox cb) {
        cb.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                checked[position] = !checked[position];
            }
        });
    }

    private View initConvertView(View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(context).inflate(R.layout.check_box_item, parent, false);
        }
        return convertView;
    }`